### ISSUE TITLE
feature request : parallelize to make it run on multiple cores

### ISSUE THREAD

**Issue by kanihal** [adrianlopezroche/fdupes#103]
I had ~125 GB of data files and it took > 1 day to complete fdupes run since it uses only single core.

I think it would be very useful if an option of choosing the number of cores to run and parallelize fudpes accordingly.

**Comment by mlybarger** [comment]
i agree.  i have lots of cpu's and would appreciate huge leaps in performance on large dirs (mp3, jpg, mp4 directories)

**Comment by quixand** [comment]
or possibly reducing the accuracy of the hashing with something like https://github.com/MatthewJohn/shamean ? to speed things up.

**Comment by FabioPedretti** [comment]
Blake3 is also a hashing function parallelizzable on multi core CPUs:
https://github.com/BLAKE3-team/BLAKE3

**Comment by adrianlopezroche** [comment]
The greatest bottleneck is disk access rather than hashing per se. Closing this as not planned.
